src/common/validators/validateBy.ts

```diff
@@ -1,16 +1,25 @@
-import { FnLike } from '@seedcompany/common';
+import { Type } from '@nestjs/common';
 import {
   registerDecorator,
   ValidationOptions,
   ValidatorConstraintInterface,
 } from 'class-validator';
+import { MergeExclusive } from 'type-fest';
 
-export interface ValidateByOptions {
-  name: string;
-  constraints?: any[];
-  validator: ValidatorConstraintInterface | FnLike;
```

Comment:
This `FnLike` should actually be a class reference (a constructor), so I converted to that.
Raw functions do not work.
`name` & `async` are given with constraint class decorator, so fixed those types as well.